src/main/java/de/tum/in/www1/artemis/service/connectors/AtheneService.java

```diff
@@ -0,0 +1,269 @@
+     * @param exercise the exercise the automatic assessments should be calculated for
+     * @param maxRetries number of retries before the request will be canceled
+     */
+    public void submitJob(TextExercise exercise, int maxRetries) {
+        log.debug("Start Athene Service for Text Exercise \"" + exercise.getTitle() + "\" (#" + exercise.getId() + ").");
+
+        // Find all submissions for Exercise
+        List<TextSubmission> textSubmissions = textSubmissionService.getTextSubmissionsByExerciseId(exercise.getId(), true, false);
+
+        // We only support english languages so far, to prevent corruption of the clustering
+        textSubmissions.removeIf(textSubmission -> textSubmission.getLanguage() != Language.ENGLISH);
+
+        // Athene only works with 10 or more submissions
+        if (textSubmissions.size() >= 10) {
+
+            log.info("Calling Remote Service to calculate automatic feedback for " + textSubmissions.size() + " submissions.");
+
+            try {
+                final Request request = new Request(exercise.getId(), textSubmissions, ARTEMIS_SERVER_URL + ATHENE_RESULT_API_PATH + exercise.getId());
+                Response response = connector.invokeWithRetry(API_ENDPOINT, request, authorizationHeaderForSymmetricSecret(API_SECRET), maxRetries);
+                log.info("Remote Service to calculate automatic feedback responded: " + response.detail);
+
+                // Register task for exercise as running, AtheneResource calls finishTask on result receive
+                startTask(exercise.getId());
+            }
+            catch (NetworkingError networkingError) {
+                log.error("Error while calling Remote Service", networkingError);
+            }
+
+        }
+    }
+
+    /**
+     * Processes results coming back from the Athene system via callbackUrl (see AtheneResource)
+     * @param clusters the Map of calculated clusters to save to the database
+     * @param blocks the list of calculated textBlocks to save to the database
+     * @param exerciseId the exercise the automatic feedback suggestions were calculated for
+     */
+    @Transactional
+    public void processResult(Map<Integer, TextCluster> clusters, List<AtheneDTO.TextBlock> blocks, Long exerciseId) {
+        log.info("Start processing incoming Athene results");
+
+        // Parse textBlocks (blocks will come as AtheneDTO.TextBlock with their submissionId and need to be parsed)
+        List<TextBlock> textBlocks = parseTextBlocks(blocks, exerciseId);
+
+        // Save textBlocks in Database
+        final Map<String, TextBlock> textBlockMap;
+        textBlockMap = textBlockRepository.saveAll(textBlocks).stream().collect(toMap(TextBlock::getId, block -> block));
+
+        // Save clusters in Database
+        processClusters(clusters, textBlockMap, exerciseId);
+
+        // Notify atheneService of finished task
+        finishTask(exerciseId);
+
+        log.info("Finished processing incoming Athene results");
```

Comment:
do we really need log.info here? And if yes, I would propose to add some additional information, e.g. which exerciseId is involved, how many results are processed, etc. otherwise the log does not really help